hazelcast/src/main/java/com/hazelcast/internal/config/AbstractConfigLocator.java

```diff
@@ -146,7 +146,7 @@ protected void loadDefaultConfigurationFromClasspath(String defaultConfigFile) {
 
     protected boolean loadConfigurationFromClasspath(String configFileName) {
         try {
-            URL url = Config.class.getClassLoader().getResource(configFileName);
+            URL url = Thread.currentThread().getContextClassLoader().getResource(configFileName);
```

Comment:
Just for the record, as we discussed in private, the default config has to be loaded from the `hazelcast.jar` always, therefore, `Config.class.getClassloader()` should be the right one here and TCCL should be used only in the other load methods.

Reply:
I tested this setup on Quarkus in _dev_ mode and works correctly 👍

Reply:
@pivovarit Could you please update the PR accordingly?

Reply:
It's done already. I think that this comment is just hanging on an outdated change. Check the file:
https://github.com/hazelcast/hazelcast/blob/c2a80de62aa33ac9eb1fe3c0a8bf2c969860b884/hazelcast/src/main/java/com/hazelcast/internal/config/AbstractConfigLocator.java#L131-L138

Reply:
Ah, I probably commented on the wrong method. This one is `loadConfigurationFromClasspath`, `loadDefaultConfigurationFromClasspath` indeed uses `Config` classloader as exoected.